packages/runtime-common/index-writer.ts

```diff
@@ -250,6 +256,57 @@ export class Batch {
     return parseInt(total);
   }
 
+  private async updateRealmMeta() {
+    let results = await this.query([
+      `SELECT CAST(count(i.url) AS INTEGER) as total, i.display_names->>0 as display_name, i.types->>0 as code_ref
+       FROM boxel_index as i
+       INNER JOIN realm_versions r ON i.realm_url = r.realm_url
+          WHERE`,
+      ...every([
+        ['i.realm_url =', param(this.realmURL.href)],
+        ['i.type = ', param('instance')],
+        ['i.types IS NOT NULL'],
+        realmVersionExpression({
+          useWorkInProgressIndex: true,
+          withMaxVersion: this.realmVersion,
+        }),
+      ]),
+      `GROUP BY i.display_names->>0, i.types->>0`,
+    ] as Expression);
+
+    let { nameExpressions, valueExpressions } = asExpressions(
+      {
+        realm_url: this.realmURL.href,
+        realm_version: this.realmVersion,
+        value: results,
+        indexed_at: unixTime(new Date().getTime()),
+      } as Omit<RealmMetaTable, 'indexed_at'> & {
+        indexed_at: number;
+      },
+      {
+        jsonFields: ['value'],
+      },
+    );
+
+    await this.query([
+      ...upsert(
+        'realm_meta',
+        'realm_meta_pkey',
+        nameExpressions,
+        valueExpressions,
+      ),
+    ]);
+
+    await this.query([
+      `DELETE FROM realm_meta`,
+      'WHERE',
+      ...every([
+        ['realm_version <', param(this.realmVersion)],
+        ['realm_url =', param(this.realmURL.href)],
+      ]),
+    ] as Expression);
```

Comment:
you are still doing this in teh wrong place. you need to add this logic _after_ line 215 in the `done()` method